otter/util/zk.py

```diff
@@ -170,3 +199,113 @@ def get_zk_dispatcher(kz_client):
             partial(perform_get_stat, kz_client),
         AcquireLock: perform_acquire_lock
     })
+
+
+class PollingLock(object):
+    """
+    Zookeeper lock recipe that polls the children on interval basis instead
+    of leaving a watch on previous child. It's supposed to replace
+    :obj:`kazoo.recipe.lock.Lock` and has same signature and _similar_ behavior
+    for ``acquire`` and ``release`` methods.
+    """
+
+    def __init__(self, dispatcher, path, identifier=""):
+        self.dispatcher = dispatcher
+        self.path = path
+        self.identifier = identifier
+        self._node = None
```

Comment:
Side effect?  Sigh, defect :).

This is a mostly beautiful and apparently correct implementation of the locking logic, which I *think* I understand.  Except for this piece.  `_node` is set when the lock is *going* to be acquired… but then it's cleared when the lock is released.  And then a bunch of tests set it directly in order to test things about that internal state, rather than driving it through the external interface.

I think the main thing that needs to be done here, for now, is to explicitly document the meaning and purpose of `_node`, since it's effectively part of the public interface, at least as far as the tests are concerned, and this is _very_ important for any future maintainer looking at this.

I struggled to think of how to deploy Active Nothing here, although I'm sure if we could figure it out the code would be the better for it.  It's almost like we need [something](https://github.com/glyph/automat/) that could track changes in state and automatically respond... :o)